### PR TITLE
Dynamic Configuration and Dynamic Configuration Service

### PR DESCRIPTION
**Pull request by wu-sheng**
# Dynamic Configuration
SkyWalking Configurations mostly are set through `application.yml` and OS system environment variables.
But some of them are supporting dynamic settings from upstream management system.

 Right now, SkyWalking supports following dynamic configurations.

 | Config Key | Value Description | Value Format Example |
|:----:|:----:|:----:|
|receiver-trace.default.slowDBAccessThreshold| Thresholds of slow Database statement, override `receiver-trace/default/slowDBAccessThreshold` of `applciation.yml`. | default:200,mongodb:50|

 This feature depends on upstream service, so it is **OFF** as default.
```yaml
configuration:
  none:
```

 ## Dynamic Configuration Service, DCS
Dynamic Configuration Service 
is a gRPC service, which requires the upstream system implemented.
The SkyWalking OAP fetches the configuration from the implementation(any system), after you open this implementation like this.

 ```yaml
configuration:
  grpc:
    # Upstream system hostname
    host: 127.0.0.1
    # Upstream system port
    port: 9555
    #period : 60 # Unit seconds, sync period. Default fetch every 60 seconds.
    #clusterName: "default" # the name of current cluster, set the name if you want to upstream system known.  
```

 ## 3rd party Configuration Center
We are welcome contributions to implement this module provider to support popular configuration center,       
such as Zookeeper, etcd, Consul, Nacos. Submit issue to discuss.

### CODE REVIEW

**Comment by peng-yongsheng** [comment]
> Config Key is: receiver-trace.default.slowDBAccessThreshold
> Description is: This feature depends on upstream service, so it is OFF as default.

@wu-sheng This key is used for the OAP server, but you said, it depends on upstream service. I have a confusion about what you said. What the upstream service is? Refers to the `Dynamic Configuration Service` section or other?

**Comment by peng-yongsheng** [comment]
Second question:
Config key is `receiver-trace.default.slowDBAccessThreshold`. This key is only used for OAP server. How to config the items of the agent side? How to seperate the key name from server side and agent side? @wu-sheng

File: oap-server/server-configuration/configuration-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        <version>6.2.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>configuration-api</artifactId>
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.skywalking</groupId>
+            <artifactId>library-module</artifactId>
+            <version>6.2.0-SNAPSHOT</version>
```
Comment (peng-yongsheng) [inline oap-server/server-configuration/configuration-api/pom.xml]: 6.2.0-SNAPSHOT -> ${project.version}

**Comment by wu-sheng** [comment]
> it depends on upstream service

Because the upstream service(DCS server side) must be online, so this feature is totally optional, and I set it OFF.

> Config key is receiver-trace.default.slowDBAccessThreshold. This key is only used for OAP server. How to config the items of the agent side? How to seperate the key name from server side and agent side?

This feature is not related to the agent side. If the receiver wants to push downstream, then they could create new keys, such as `register-receiver.default.agent.downstream.xxxx` if you like. But basically, this feature is designed for OAP server mostly.

File: oap-server/server-configuration/configuration-api/pom.xml (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-configuration/configuration-api/pom.xml]: Done.

**Pull request merged**